API/Controllers/CategoryController.cs

```diff
@@ -66,19 +66,36 @@ public async Task<ActionResult<IEnumerable<CategoryDto>>> GetUserCategories()
             var categories = await _categoryRepository.GetCategoriesAsync(userId);
             return Ok(categories);
         }
+
+        //[HttpGet("user={id}")]
+        //public async Task<ActionResult<IEnumerable<CategoryDto>>> GetUserCategories(int id)
+        //{
+        //    var categoryToDeleteAndItsChildrens = await _categoryRepository.
+        //        GetCategoryToDeleteAndChildrenCategories(id);
+        //    var x = 5;
+        //    var z = 9;
+        //    return Ok(categoryToDeleteAndItsChildrens);
+        //}
+
         [HttpDelete("deleteCategoryId={deleteCategoryId}")]
         public async Task<ActionResult> DeleteCategory(int deleteCategoryId)
         {
             var userId = User.GetUserId();
-            var categoryToDelete = await _categoryRepository.GetCategoryAsync(deleteCategoryId);
-            if (categoryToDelete.AppUserId != userId) return NotFound("This category doesnt belong to this user");
-            var childrenCategories = await _categoryRepository.GetChildrenCategories(categoryToDelete.Id);
-            var doesHaveParent = categoryToDelete.ParentCategoryId;
-            await _categoryRepository.DeleteCategoryAsync(categoryToDelete);
+            var categoryToDeleteAndItsChildrens = await _categoryRepository.
+                GetCategoryToDeleteAndChildrenCategories(deleteCategoryId);
+
+            var catergoryToDeleteParentId = categoryToDeleteAndItsChildrens.ParentCategoryId;
+
+            if (categoryToDeleteAndItsChildrens.AppUserId != userId) return NotFound("This category doesnt belong to this user");
+            await _categoryRepository.DeleteCategoryAsync(categoryToDeleteAndItsChildrens);
 
-            if (childrenCategories != null && doesHaveParent != null)
+            if (categoryToDeleteAndItsChildrens.ChildCategories != null && catergoryToDeleteParentId != null)
```

Comment:
nullem to to raczej nigdy nie będzie, co najwyżej może być puste, możesz sprawdzić czy nie jest puste